src/test/java/guitests/guihandles/TaskCardViewHandle.java

```diff
@@ -0,0 +1,326 @@
+            return !getDateBoxVisibility();
+        } else if (startTime.isPresent() && endTime.isPresent()) {
+            //When start and end date are available, expect event format
+            expectedDateText = timeUtil.getEventTimeText(startTime.get(), endTime.get());
+        } else if (endTime.isPresent()) {
+            //When only end time is present, expect deadline format
+            expectedDateText = timeUtil.getTaskDeadlineText(endTime.get());
+        } else {
+            //Otherwise, illegal date state.
+            throw new IllegalStateException("Start time is present, but end time is not.");
+        }
+        return expectedDateText.equals(displayedDateText);
+    }
+
+    private boolean isLocationCorrect(ImmutableTask task) {
+        java.util.Optional<String> location = task.getLocation();
+
+        if (location.isPresent()) {
+            //If there is a location, it should match with the displayed location.
+            String expected = location.get();
+            String actual = getDisplayedLocation();
+            return expected.equals(actual);
+        } else {
+            //Description should be hidden when there is no description.
+            boolean expected = false;
+            boolean actual = getLocationBoxVisibility();
+            return expected == actual;
+        }
+    }
+
+    private boolean isPinDisplayCorrect(ImmutableTask task) {
+        boolean expected = task.isPinned();
+        boolean actual = getPinImageVisibility();
+        return expected == actual;
+    }
+
+    private boolean isCompletedDisplayCorrect(ImmutableTask task) {
+        boolean expected = task.isCompleted();
+        boolean actual = isCompletedStyleApplied();
+        return expected == actual;
+    }
+
+    private boolean isOverdueDisplayCorrect(ImmutableTask task) {
+        java.util.Optional<LocalDateTime> endTime = task.getEndTime();
+        boolean actual = isOverdueStyleApplied();
+        boolean expected;
+
+        if (endTime.isPresent()) {
+            expected = seedu.todo.testutil.TimeUtil.isOverdue(endTime.get());
+        } else {
+            expected = false;
+        }
+        return expected == actual;
+    }
+
+    private boolean isTypeDisplayCorrect(ImmutableTask task) {
+        String actual = getDisplayedTypeLabel();
+        String expected;
+
+        if (task.isEvent()) {
+            expected = "Event";
+        } else {
+            expected = "Task";
+        }
+
+        return expected.equals(actual);
+    }
+
+    /* View Elements Helper Methods */
+    /**
+     * Search and returns exactly one matching node.
+     *
+     * @param fieldId Field ID to search inside the parent node.
+     * @return Returns one appropriate node that matches the {@code fieldId}.
+     * @throws NullPointerException when no node with {@code fieldId} can be found, intentionally breaking the tests.
+     */
+    @Override
+    protected Node getNode(String fieldId) throws NullPointerException {
```

Comment:
`Optional` already throws an exception if you try to call `get` on it when there is no value present.